src/docfx/lib/ProcessUtility.cs

```diff
@@ -16,6 +18,147 @@ namespace Microsoft.Docs.Build
     /// </summary>
     internal static class ProcessUtility
     {
+        private const int _defaultLockExpireTimeInSecond = 60 * 20;
+
+        /// <summary>
+        /// Acquire a shared lock for input lock name
+        /// The returned `acquirer` are used for tracking the acquired lock, instead of thread info, since the thread info may change in asynchronous programming model
+        /// </summary>
+        public static async Task<(bool acquired, string acquirer)> AcquireSharedLock(string lockName)
```

Comment:
just return a bool state makes it harder for others to use this lock, caller have to wait and retry. There are 2 patterns in .NET: [wait with a timeout](https://docs.microsoft.com/en-us/dotnet/api/system.threading.waithandle.waitone?view=netframework-4.7.2#System_Threading_WaitHandle_WaitOne_System_Int32_) or [TryEnter](https://docs.microsoft.com/en-us/dotnet/api/system.threading.readerwriterlockslim.tryenterreadlock?view=netframework-4.7.2)

Reply:
current requirement doesn't need wait, we can do that later if you have the requirments,